lib/computer_brain.rb

```diff
@@ -1,14 +1,101 @@
 require 'pry'
 
 class ComputerBrain
-  attr_accessor :attacked_keys, :available_keys
+  attr_accessor :attacked_keys, :available_keys, :validated_keys
 
-  def initialize(user_board)
+  def initialize(user_board, cpu_board)
     @user_board = user_board
+    @cpu_board = cpu_board
     @attacked_keys = []
     @available_keys = @user_board.cells.keys
+    @validated_keys = []
   end
 
+  ##Computer place methods
+
+  def make_limited_placement_array(ship)
+    length = ship.length
+    max_width = @cpu_board.width - length + 1
+    max_height = (@cpu_board.height - length + 65).chr
+
+    potential_first_keys = []
+    ("A"..max_height).to_a.each do |letter|
+      (1..max_width).to_a.each do |number|
+        potential_first_keys << "#{letter}#{number}"
+      end
+    end
+    return potential_first_keys
+  end
+
+  def directional_search(coordinate, direction, ship)
+    letter = coordinate.split("")[0]
+    number = coordinate.split("")[1].to_i
+    potential_keys = []
+
+    if direction == :vertical
+      end_letter = (letter.ord + ship.length - 1).chr
+      test_range = (letter..end_letter).to_a
+      test_range.each do |dyn_letter|
+        potential_keys << "#{dyn_letter}#{number}"
+      end
+      if valid_potential_keys?(potential_keys) == true
+        return potential_keys
+      else
+        return false
+      end
+    elsif direction == :horizontal
+      end_number = number + ship.length - 1
+      test_range = (number..end_number).to_a
+      test_range.each do |dyn_number|
+        potential_keys << "#{letter}#{dyn_number}"
+      end
+      if valid_potential_keys?(potential_keys) == true
```

Comment:
Also, seeing some repetition between lines 41 and 52, which might a chance for another small method!

Reply:
Those lines are the same but they are already calling another method. You could remove the "== true" and it wouldn't look as repetitive but I like those for clarity.